djitellopy/tello.py

```diff
@@ -415,30 +409,38 @@ def send_command_with_return(self, command: str, timeout: int = RESPONSE_TIMEOUT
         Return:
             bool/str: str with response text on success, False when unsuccessfull.
         """
-        # Commands very consecutive makes the drone not respond to them. So wait at least self.TIME_BTW_COMMANDS seconds
+        # Commands very consecutive makes the drone not respond to them.
+        # So wait at least self.TIME_BTW_COMMANDS seconds
         diff = time.time() - self.last_received_command_timestamp
         if diff < self.TIME_BTW_COMMANDS:
-            self.LOGGER.debug('Waiting {} seconds to execute command {}...'.format(diff, command))
-            time.sleep(diff)
+            self.LOGGER.debug('Waiting {} seconds to execute command: {}...'.format(diff, command))
+            self.sleep(diff)
 
-        self.LOGGER.info('Send command: ' + command)
+        self.LOGGER.info("Send command: '{}'".format(command))
         timestamp = time.time()
 
         client_socket.sendto(command.encode('utf-8'), self.address)
 
         responses = self.get_own_udp_object()['responses']
-        while len(responses) == 0:
+
+        while not responses:
             if time.time() - timestamp > timeout:
-                self.LOGGER.warning('Timeout exceed on command ' + command)
-                return "Timeout error!"
-            else:
-                time.sleep(0.1)
+                message = "Aborting command '{}'. Return after timeout of {}s".format(command, timeout)
+                self.LOGGER.warning(message)
+                return message
+            self.sleep(0.1)  # Sleep during send command
 
         self.last_received_command_timestamp = time.time()
-        response = responses.pop(0)
-        response = response.decode('utf-8').rstrip("\r\n")
 
-        self.LOGGER.info('Response {}: {}'.format(command, response))
+        first_response = responses.pop(0)  # first datum from socket
+        try:
+            response_decoded = first_response.decode('utf-8').rstrip("\r\n")
+        except UnicodeDecodeError as e:
```

Comment:
I don't want to do a functional change in this PR, but if I keep using it I can think about what works here.
The reason I add the try here is that I got a complaint when using it the first time (but only the first time). I'm running it from a Mac and use the 100$ small drone.
I went for print here since I don't know which message level it should be (I can do warning, but I don't know if this didn't kill the run altogether. Was there a log fatal with the logger?)